app/models/dispatch.rb

```diff
@@ -24,9 +24,15 @@
 #
 
 class Dispatch < ApplicationRecord
+    include Postable 
+
     acts_as_paranoid 
 
     belongs_to :fdp 
     
     has_many :dispatch_items
+
+    after_save :pre_post
+
+    after_update :reverse
```

Comment:
I think we should do `:patch` here instead. Which is reversal followed by a new `Posting`.